examples/energy-management-app/energy-management-common/include/DeviceEnergyManagementDelegateImpl.h

```diff
@@ -18,69 +18,276 @@
 {
 public:
+    DeviceEnergyManagementDelegate();
+
+    void SetDeviceEnergyManagementInstance(DeviceEnergyManagement::Instance & instance);
+
+    void SetDEMManufacturerDelegate(DEMManufacturerDelegate & deviceEnergyManagementManufacturerDelegate);
+
+    /**
+     *
+     * Implement the DeviceEnergyManagement::Delegate interface
+     *
+     */
+
+    /**
+     * @brief Implements a handler to begin to adjust client power
+     *        consumption/generation to the level requested.
+     *
+     * @param power Milli-Watts the ESA SHALL use during the adjustment period.
+     * @param duration The duration that the ESA SHALL maintain the requested power for.
+     * @return  Success if the adjustment is accepted; otherwise the command SHALL be rejected with appropriate error.
+     */
     virtual Status PowerAdjustRequest(const int64_t power, const uint32_t duration, AdjustmentCauseEnum cause) override;
+
+    /**
+     * @brief Make the ESA end the active power adjustment session & return to normal (or idle) power levels.
+     *        The ESA SHALL also generate an PowerAdjustEnd Event and the ESAState SHALL be restored to Online.
+     *
+     * @return It should report SUCCESS if successful and FAILURE otherwise.
+     */
     virtual Status CancelPowerAdjustRequest() override;
+
+    /**
+     * @brief The ESA SHALL update its Forecast attribute with the RequestedStartTime including a new ForecastID.
+     *
+     *   If the ESA supports ForecastAdjustment, and the ESAState is not UserOptOut and the RequestedStartTime is after
+     *   the EarliestStartTime and the resulting EndTime is before the LatestEndTime, then ESA SHALL accept the request
+     *   to modify the Start Time.
+     *   A client can estimate the entire Forecast sequence duration by computing the EndTime - StartTime fields from the
+     *   Forecast attribute, and therefore avoid scheduling the start time too late.
+     *
+     * @param requestedStartTime The requested start time in UTC that the client would like the appliance to shift its power
+     * forecast to.
+     * @param cause    Who (Grid/local) is triggering this change.
+     *
+     * @return Success if the StartTime in the Forecast is updated, otherwise the command SHALL be rejected with appropriate
+     * IM_Status.
+     */
     virtual Status StartTimeAdjustRequest(const uint32_t requestedStartTime, AdjustmentCauseEnum cause) override;
+
+    /**
+     * @brief Handler for PauseRequest command
+     *
+     *   If the ESA supports FA and the SlotIsPauseable field is true in the ActiveSlotNumber
+     *   index in the Slots list, and the ESAState is not UserOptOut then the ESA SHALL allow its current
+     *   operation to be Paused.
+     *
+     *   During this state the ESA SHALL not consume or produce significant power (other than required to keep its
+     *   basic control system operational).
+     *
+     * @param duration Duration that the ESA SHALL be paused for.
```

Comment:
duration needs units. Will stop commenting on units, just noting that there are several instances.

Reply:
As above - duration is from the spec name which doesn’t include _s etc

Reply:
In C++ code I expect units on variables